locators/patient-master.ts

```diff
@@ -15,20 +22,121 @@ export class PatientMaster {
     this.table = this.page.locator(
       "(//*[@class='table-wrapper'])/div/div/div/div/div/div/table"
     );
+    this.heading = this.page.getByText("Patient Master");
+    this.search = this.page.locator("//*[@data-testid='search-filter-input']");
+    this.sear_result = this.page.locator(
+      "//tbody[@class ='ant-table-tbody']/child::tr/td[2]"
+    );
+    // this.search_key = "James";
+    this.filter = this.page.locator("//*[@data-testid='down-arrow']");
+    this.criteria_dropdown = this.page.locator(
+      "//*[@data-testid='criteria-dropdown']"
+    );
+    this.select_dropdown = this.page.locator(
+      "//div[@data-testid='criteria-value-dropdown']"
+    );
   }
 
-  async searchWithProvidedText(text: string) {
-    await this.searchField.fill(text);
+  async clickOnCreateButton() {
+    await this.createButton.click();
   }
-
   async validateCreatedPatient(text: string) {
     await this.searchField.fill(text);
     await this.page.waitForTimeout(3000);
     let tableData = await this.table.innerText();
     expect(tableData).toContain(text);
   }
 
-  async clickOnCreateButton() {
-    await this.createButton.click();
+  async search_enter(search_key) {
+    this.search_key_recieved = search_key;
+    await this.search.fill(this.search_key_recieved);
+    await this.page.waitForTimeout(10000);
+  }
+
+  async verify_search_result() {
+    let text = await this.sear_result.innerText();
+    await expect(text).toEqual(
+      expect.stringContaining(this.search_key_recieved)
+    );
+  }
+
+  async click_on_filters() {
+    await this.filter.click();
+  }
+
+  async click_on_criteria_dropdown() {
+    await this.criteria_dropdown.click();
+  }
+
+  async verify_criteria_dropdown() {
+    let dropdown_expected = [
```

Comment:
can we move this to class variables?

Reply:
These variables will not be  used anywhere else - adding it in class variable would instantiate these variables also - everytime - hence added this in methods only

Reply:
This might get used while applying filters. It is fine if that is not the case.

Reply:
Have kept it as of now the same